bleak/backends/winrt/client.py

```diff
@@ -0,0 +1,836 @@
+                asyncio.create_task(callback(s, d))
+
+        else:
+            bleak_callback = callback
+
+        if not isinstance(char_specifier, BleakGATTCharacteristic):
+            characteristic = self.services.get_characteristic(char_specifier)
+        else:
+            characteristic = char_specifier
+        if not characteristic:
+            raise BleakError("Characteristic {0} not found!".format(char_specifier))
+
+        if self._notification_callbacks.get(characteristic.handle):
+            await self.stop_notify(characteristic)
+
+        characteristic_obj = characteristic.obj
+        if (
+            characteristic_obj.characteristic_properties
+            & GattCharacteristicProperties.INDICATE
+        ):
+            cccd = GattClientCharacteristicConfigurationDescriptorValue.INDICATE
+        elif (
+            characteristic_obj.characteristic_properties
+            & GattCharacteristicProperties.NOTIFY
+        ):
+            cccd = GattClientCharacteristicConfigurationDescriptorValue.NOTIFY
+        else:
+            cccd = GattClientCharacteristicConfigurationDescriptorValue.NONE
+
+        fcn = _notification_wrapper(bleak_callback, asyncio.get_event_loop())
+        fcn_token = characteristic_obj.add_value_changed(fcn)
+        self._notification_callbacks[characteristic.handle] = fcn, fcn_token
+        status = await characteristic_obj.write_client_characteristic_configuration_descriptor_async(
+            cccd
+        )
+
+        if status != GattCommunicationStatus.SUCCESS:
+            # This usually happens when a device reports that it support indicate,
+            # but it actually doesn't.
+            if characteristic.handle in self._notification_callbacks:
+                callback, token = self._notification_callbacks.pop(
+                    characteristic.handle
+                )
+                characteristic_obj.remove_value_changed(token)
+
+            raise BleakError(
+                "Could not start notify on {0}: {1}".format(
+                    characteristic.uuid, _communication_statues.get(status, "")
+                )
+            )
+
+    async def stop_notify(
+        self, char_specifier: Union[BleakGATTCharacteristic, int, str, uuid.UUID]
+    ) -> None:
+        """Deactivate notification/indication on a specified characteristic.
+
+        Args:
+            char_specifier (BleakGATTCharacteristic, int, str or UUID): The characteristic to deactivate
+                notification/indication on, specified by either integer handle, UUID or
+                directly by the BleakGATTCharacteristic object representing it.
+
+        """
+        if not isinstance(char_specifier, BleakGATTCharacteristic):
+            characteristic = self.services.get_characteristic(char_specifier)
+        else:
+            characteristic = char_specifier
+        if not characteristic:
+            raise BleakError("Characteristic {} not found!".format(char_specifier))
```

Comment:
`f""`? Here and many other places

Reply:
I am planing on doing a separate PR with f-strings later on after applying [flynt](https://github.com/ikamensh/flynt) on the codebase.

Reply:
nice